tests/templates/test_layers.py

```diff
@@ -298,64 +298,64 @@ def impr(self, request):
-        n_rots = 1
-        n_wires = 2
-        dev = qml.device("default.qubit", wires=n_wires)
-        weights = np.random.randn(n_layers, n_rots)
+    def test_different_circuits_for_different_seeds(self, tol):
+        """Test that RandomLayers() does not necessarily have the same output for two different seeds."""
+        dev = qml.device("default.qubit", wires=2)
+        weights = [[0.1, 0.2, 0.3]]
 
         def circuit1(weights):
-            RandomLayers(weights=weights, wires=range(n_wires))
+            RandomLayers(weights=weights, wires=range(2), seed=1)
             return qml.expval(qml.PauliZ(0))
 
         def circuit2(weights):
-            RandomLayers(weights=weights, wires=range(n_wires))
+            RandomLayers(weights=weights, wires=range(2), seed=2)
             return qml.expval(qml.PauliZ(0))
 
         qnode1 = qml.QNode(circuit1, dev)
         qnode2 = qml.QNode(circuit2, dev)
 
-        assert np.allclose(qnode1(weights), qnode2(weights), atol=tol)
+        assert not np.allclose(qnode1(weights), qnode2(weights), atol=tol)
 
-    def test_random_layers_two_seeds_different(self, n_layers, tol):
-        """Test that RandomLayers() does not have the same output for two different seeds."""
-        n_rots = 10
-        n_wires = 2
-        dev = qml.device("default.qubit", wires=n_wires)
-        weights = np.random.randn(n_layers, n_rots)
+    def test_same_circuit_in_each_mutable_qnode_call(self, tol):
+        """Test that RandomLayers() creates the same circuit in two calls of a mutable qnode."""
+        dev = qml.device("default.qubit", wires=2)
+        weights = [[0.1, 0.2, 0.3]]
 
-        def circuit1(weights):
-            RandomLayers(weights=weights, wires=range(n_wires), seed=0)
+        @qml.qnode(dev, mutable=True)
+        def circuit(weights):
+            RandomLayers(weights=weights, wires=range(2))
             return qml.expval(qml.PauliZ(0))
 
-        def circuit2(weights):
-            RandomLayers(weights=weights, wires=range(n_wires), seed=1)
-            return qml.expval(qml.PauliZ(0))
+        first_call = circuit(weights)
+        second_call = circuit(weights)
+        assert np.allclose(first_call, second_call, atol=tol)
 
-        qnode1 = qml.QNode(circuit1, dev)
-        qnode2 = qml.QNode(circuit2, dev)
+    def test_same_circuit_in_each_nonmutable_qnode_call(self, tol):
+        """Test that RandomLayers() creates the same circuit in two calls of a nonmutable qnode."""
+        dev = qml.device("default.qubit", wires=2)
+        weights = [[0.1, 0.2, 0.3]]
 
-        assert not np.allclose(qnode1(weights), qnode2(weights), atol=tol)
+        @qml.qnode(dev, mutable=False)
+        def circuit(weights):
+            RandomLayers(weights=weights, wires=range(2))
+            return qml.expval(qml.PauliZ(0))
+
+        first_call = circuit(weights)
+        second_call = circuit(weights)
+        assert np.allclose(first_call, second_call, atol=tol)
```

Comment:
Could possibly also have a test that immutable circuits with `seed=np.random.randint(100)` have the same output (i.e., the last bit of Usage Details).

Reply:
I changed this and now test the ``seed=None`` case.

The problem is that the test is using randomness, which one should not do. To make it very unlikely that the circuit constructed with different seeds is the same (which can happen), I use 100 rotation gates (where CNOTs are randomly inserted between them. 

Or shall I just not test this case to avoid random tests?

Reply:
Oh that is a good point :thinking:

Yes maybe we should avoid this test.

Reply:
Ok, I deleted the test for the mutable qnode. In a sense, if qnode does what it is supposed to, and random.seed works well, then this should make testing this case unnecessary!